spec/patcher_spec.rb

```diff
@@ -26,6 +26,27 @@ def get_patcher(filename, logging: true)
     expect(get_patcher('pie.elf').interpreter).to eq '/lib64/ld-linux-x86-64.so.2'
   end
 
+  describe 'initializer arguments' do
+    it 'doesnt accept both logging and on_error' do
+      expect { get_patcher('rpath.elf', logging: true, on_error: :silent) }.to raise_error(ArgumentError)
+    end
+
+    it 'accepts one of [:log :silent :exception] as on_error value' do
+      accepted_syms = %i[log exception silent]
+      accepted_syms.each do |on_error|
+        expect { get_patcher('rpath.elf', on_error: on_error) }.not_to raise_error
+      end
+      expect { get_patcher('rpath.elf', on_error: :nyan) }.to raise_error(ArgumentError)
+    end
+
+    it 'returns nil on_error :silent' do
+      expect(get_patcher('libtest.so', on_error: :silent).interpreter).to be_nil
+      expect(get_patcher('runpath.elf', on_error: :silent).rpath).to be_nil
+      expect(get_patcher('rpath.elf', on_error: :silent).runpath).to be_nil
+      expect(get_patcher('static.elf', on_error: :silent).soname).to be_nil
+    end
+  end
```

Comment:
You are good at writing specs, I'm appreciate it

Reply:
i removed the spec for "doesn't accept both logging and on_error".
We are doing half hearted checks in initialize, 
if u wish we can spec the existing behaviour.